client/src/components/chess-square/ChessSquare.js

```diff
@@ -1,5 +1,52 @@
+/* eslint-disable react/require-default-props */
 import React from 'react';
+import PropTypes from 'prop-types';
+import './ChessSquare.css';
+import chessPieces from './chess-pieces';
 
-const ChessSquare = () => (<div>ChessSquare Component</div>);
+const ChessSquare = ({
+  isDark, gridPosition, pieceType, pieceColor, highlight, handleSelect,
+}) => {
+  let piece = '';
+
+  if (pieceType && pieceColor) {
+    const key = `${pieceType}${pieceColor}`;
+    piece = chessPieces[key];
+  }
+
+  let squareClass = isDark ? 'dark-square' : 'light-square';
+  squareClass += highlight ? ' square-highlight' : '';
+
+  const handleSqClick = () => {
```

Comment:
Create this function outside render as this will be created on every render.

Reply:
Done